### PR TITLE
Messaging API - Message actions route seen/unseen

### PR DESCRIPTION
**Pull request by Thurfjell**
### Ticket:

- [XXX](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/XXX)

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Dev change**
- [ ] **Additional tests**
- [ ] **Documentation**
- [ ] **Other**

### Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

File: apps/messaging-api/src/routes/message-actions/index.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            isSeen: boolean;
+          }>
+        | undefined;
+
+      const jobEvents: { type: EventType; data: MessageEventData }[] = [];
+      let poolClient: PoolClient | undefined;
+      try {
+        poolClient = await app.pg.pool.connect();
+        console.log("broder tuck");
+        console.log(userId, req.body.messageId);
```
Comment (SamSalvatico) [inline apps/messaging-api/src/routes/message-actions/index.ts:76]:
```suggestion
```

File: apps/messaging-api/src/routes/message-actions/index.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        updateQueryResult = await poolClient.query<{ isSeen: boolean }>(
+          `
+            with message_selection as(
+                select m.id from messages m
+                join users u on (u.user_profile_id = $1 or u.id::text = $1)
+                where 
+                    (u.id::text = $1 or u.user_profile_id = $1)
+                    and is_delivered = true 
+                    and m.id = $2
+                    and $3 != m.is_seen
```
Comment (SamSalvatico) [inline apps/messaging-api/src/routes/message-actions/index.ts:105]: Do we want to return an error if an already seen message is set as seen again?

**Review by SamSalvatico — COMMENTED** [review]
Few comments, but great work

File: apps/messaging-api/src/routes/message-actions/index.ts (hunk 2 above)
Comment (Thurfjell) [reply apps/messaging-api/src/routes/message-actions/index.ts:103]: No with motivation:

(big) if we add more actions, we'll treat this as an ordinary put. There's nothing inherently criminal with putting a resource to the same. Just nothing happens. I think this is normal behaviour.

Comment (SamSalvatico) [reply apps/messaging-api/src/routes/message-actions/index.ts:103]: Sorry, I misread the code totally fine

**Review by SamSalvatico — APPROVED** [review]

**Pull request merged**